### ISSUE TITLE
No 'Access-Control-Allow-Origin' header is present on the requested resource. (When not using localhost)

### ISSUE THREAD

**Issue by Steague** [okta/okta-oidc-js#354]
## I'm submitting this issue for the package(s):

- [ ] jwt-verifier
- [ ] okta-angular
- [ ] oidc-middleware
- [X] okta-react
- [ ] okta-react-native
- [ ] okta-vue

## I'm submitting a:

- [X] Bug report  
- [ ] Feature request
- [ ] Other (Describe below)

## Current behavior

Receiving browser console error message when attempting to login when not using localhost a domain.

```Access to XMLHttpRequest at 'https://{MY ORG}.oktapreview.com/api/v1/authn' from origin 'http://{AWS DOMAIN}.amazonaws.com' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource.```

## Expected behavior

Valid CORS headers should be sent back for domains other than localhost.

## Minimal reproduction of the problem with instructions

- Follow your tutorial here: https://developer.okta.com/code/react/okta_react
- Deploy the app to somewhere (like Amazon AWS)
- Attempt to login
- Observe browser console for CORS error message

## Extra information about the use case/user story you are trying to implement

None

## Environment

- @okta/okta-react Package Version: 1.1.4
- @okta/okta-auth-js Package Version: 2.2.0
- Browser: Chrome
- OS: MacOS
- Node version (`node -v`): 8.11
- Other:

**Comment by webberpuma** [comment]
I experienced the similar issue too
starting from today, the login failed and the browser console returned the following error:

Access to XMLHttpRequest at 'https://dev-{xxxxxx}.oktapreview.com/oauth2/default/v1/keys' from origin 'http://localhost:3000' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: The value of the 'Access-Control-Allow-Origin' header in the response must not be the wildcard '*' when the request's credentials mode is 'include'. The credentials mode of requests initiated by the XMLHttpRequest is controlled by the withCredentials attribute.

strange enough, the app (both local or in a live server) ran without this problem just a few days ago and I didn't change any code! but starting from these two days, I received this error which prevented me to login

**Comment by Kamran1122** [comment]
Had a similar issue today. Downgrading to 1.0.2 worked for me.

**Comment by Kamran1122** [comment]
Thanks @snanduri-okta the issue was fixed.

**Comment by jmelberg-okta** [comment]
Closing this issue as it is a duplicate of #337

**Comment by jeff-ofobrukweta** [comment]
please am using keycloak and have same issues when i already added header with bearer-token
import axios from 'axios';
import {keycloak} from '../keycloak-config';

export default axios.create({
  baseURL: `http://localhost:5000/api/v1`,
  headers: {Authorization: 'bearer '+ keycloak.idToken}
});

//Thunk
export const Allvouchers =  () => {
    merchantId = keycloak.idTokenParsed.sub
    const request =  API.get(`/all?merchantId=${merchantId}`);
    return (dispatch)=>{
        dispatch(getAllVouchers());
        request
        .then(vouchers =>{
            dispatch(getVouchersSuccess(vouchers.data));
        }).catch((error) => {
            dispatch(getVouchersFailure(error.response || 'Problem getting  vouchers'));
        })
    };
}

**Comment by jeff-ofobrukweta** [comment]
error is:
xhr.js:173 GET http://localhost:5000/api/v1/all?merchantId=6f694a7b-20a6-4847-8abe-7ecddae7b1ba 500 (Internal Server Error)
dispatchXhrRequest @ xhr.js:173
xhrAdapter @ xhr.js:18
dispatchRequest @ dispatchRequest.js:49
Promise.then (async)
request @ Axios.js:55
Axios.(anonymous function) @ Axios.js:65
wrap @ bind.js:11
Allvouchers @ voucherThunk.js:31
Allvouchers @ Vouchers.All.js:147
componentDidMount @ Vouchers.All.js:61
commitLifeCycles @ react-dom.development.js:16998
commitAllLifeCycles @ react-dom.development.js:18512
callCallback @ react-dom.development.js:147
invokeGuardedCallbackDev @ react-dom.development.js:196
invokeGuardedCallback @ react-dom.development.js:250
commitRoot @ react-dom.development.js:18717
completeRoot @ react-dom.development.js:20247
performWorkOnRoot @ react-dom.development.js:20170
performWork @ react-dom.development.js:20075
performSyncWork @ react-dom.development.js:20049
requestWork @ react-dom.development.js:19904
scheduleWork @ react-dom.development.js:19711
enqueueSetState @ react-dom.development.js:12936
push../node_modules/react/cjs/react.development.js.Component.setState @ react.development.js:356
(anonymous) @ dashboard.js:192
setTimeout (async)
(anonymous) @ dashboard.js:192
setSuccess @ keycloak.js:1036
(anonymous) @ keycloak.js:133
setSuccess @ keycloak.js:1036
authSuccess @ keycloak.js:619
req.onreadystatechange @ keycloak.js:596
XMLHttpRequest.send (async)
processCallback @ keycloak.js:605
(anonymous) @ keycloak.js:192
setSuccess @ keycloak.js:1036
iframe.onload @ keycloak.js:1096
load (async)
setupCheckLoginIframe @ keycloak.js:1087
processInit @ keycloak.js:191
success @ keycloak.js:1050
Keycloak.kc.init @ keycloak.js:238
componentDidMount @ dashboard.js:184
commitLifeCycles @ react-dom.development.js:16998
commitAllLifeCycles @ react-dom.development.js:18512
callCallback @ react-dom.development.js:147
invokeGuardedCallbackDev @ react-dom.development.js:196
invokeGuardedCallback @ react-dom.development.js:250
commitRoot @ react-dom.development.js:18717
completeRoot @ react-dom.development.js:20247
performWorkOnRoot @ react-dom.development.js:20170
performWork @ react-dom.development.js:20075
performSyncWork @ react-dom.development.js:20049
requestWork @ react-dom.development.js:19904
scheduleWork @ react-dom.development.js:19711
scheduleRootUpdate @ react-dom.development.js:20415
updateContainerAtExpirationTime @ react-dom.development.js:20441
updateContainer @ react-dom.development.js:20509
push../node_modules/react-dom/cjs/react-dom.development.js.ReactRoot.render @ react-dom.development.js:20820
(anonymous) @ react-dom.development.js:20974
unbatchedUpdates @ react-dom.development.js:20292
legacyRenderSubtreeIntoContainer @ react-dom.development.js:20970
render @ react-dom.development.js:21037
./src/index.js @ index.js:44
__webpack_require__ @ bootstrap:782
fn @ bootstrap:150
0 @ navigationbar.js:19
__webpack_require__ @ bootstrap:782
checkDeferredModules @ bootstrap:45
webpackJsonpCallback @ bootstrap:32
(anonymous) @ main.chunk.js:1
history:1 Access to XMLHttpRequest at 'http://localhost:5000/api/v1/all?merchantId=6f694a7b-20a6-4847-8abe-7ecddae7b1ba' from origin 'http://localhost:3000' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.

**Comment by smitpatelx** [comment]
Guys I have a solution that might work for your local dev environment https://medium.com/@smitbpatelx/how-to-avoid-cors-policy-by-browsers-client-side-92dab755a09a
For permanent fix you can use proxy
